src/test/java/com/google/sps/AbuseDetectionTest.java

```diff
@@ -0,0 +1,232 @@
+    
+    inputTime = LocalTime.parse("23:59:55.070");
+    value = instance.addRequest(inputTime);
+        
+    inputTime = LocalTime.parse("23:59:55.090");
+    value = instance.addRequest(inputTime);
+    
+    inputTime = LocalTime.parse("23:59:55.124");
+    value = instance.addRequest(inputTime);
+
+    inputTime = LocalTime.parse("23:59:55.145");
+    value = instance.addRequest(inputTime);
+    
+    inputTime = LocalTime.parse("23:59:55.231");
+    value = instance.addRequest(inputTime);
+
+    inputTime = LocalTime.parse("23:59:55.271");
+    value = instance.addRequest(inputTime);
+
+    inputTime = LocalTime.parse("23:59:55.344");
+    value = instance.addRequest(inputTime);
+    
+    inputTime = LocalTime.parse("23:59:55.421");
+    value = instance.addRequest(inputTime);
+    
+    inputTime = LocalTime.parse("23:59:55.560");
+    value = instance.addRequest(inputTime);
+
+    inputTime = LocalTime.parse("23:59:55.590");
+    value = instance.addRequest(inputTime);
+    
+    inputTime = LocalTime.parse("23:59:55.612");
+    value = instance.addRequest(inputTime);
+
+    inputTime = LocalTime.parse("23:59:55.642");
+    value = instance.addRequest(inputTime);
+   
+    inputTime = LocalTime.parse("23:59:55.723");
+    value = instance.addRequest(inputTime);
+
+    inputTime = LocalTime.parse("23:59:55.800");
+    value = instance.addRequest(inputTime);
+
+    inputTime = LocalTime.parse("23:59:55.900");
+    value = instance.addRequest(inputTime);
+   
+    return value;
+  }
+
+  @Test
+  public void addingOneTimeRequest() {
+    AbuseDetection instance = new AbuseDetection(1, 10);
+    LocalTime inputTime = LocalTime.parse("18:19:20.010");
+    boolean value = instance.addRequest(inputTime);
+
+    Assert.assertEquals(true, value);
+  }
+
+  @Test
+  public void addingMultipleRequestsGetFalse() {
+    AbuseDetection instance = new AbuseDetection(1, 10);
+    boolean returnValue = addMultipleRequests(instance);
+    LocalTime inputTime = LocalTime.parse("18:19:20.901");
+    boolean value = instance.addRequest(inputTime);
+
+    Assert.assertEquals(false, value);
+  }
+
+  @Test
+  public void addingMultipleRequestsGetTrue() {
+    AbuseDetection instance = new AbuseDetection(1, 10);
+    boolean returnValue = addMultipleRequests(instance);
+    LocalTime inputTime = LocalTime.parse("18:19:21.012");
+    boolean value = instance.addRequest(inputTime);
+
+    Assert.assertEquals(true, value);
+  }
+
+  @Test
+  public void requestOverSecondReturnsFalse() {
+    AbuseDetection instance = new AbuseDetection(1, 10);
+    boolean returnValue = addMultipleRequests(instance);
+    LocalTime inputTime1 = LocalTime.parse("18:19:21.012");
+    boolean curVal = instance.addRequest(inputTime1);
+    LocalTime otherTime = LocalTime.parse("18:19:21.109");
+    boolean value = instance.addRequest(otherTime);
+
+    Assert.assertEquals(false, value);
+  }
+  
+
+  @Test
+  public void requestOverSecondReturnsTrue() {
```

Comment:
What is the expected behavior if we send 20 requests in the first interval (e.g. 18:19:21.000), 0 requests in the second interval (e.g. 18:19:22.000), and 20 requests in the third interval (e.g. 18:19:23.000)?

Reply:
One question for this situation,

what would the time Interval be for how many requests are being allowed in the specific timeValue in seconds. 
Would it be 20 requests in 2 seconds?

Reply:
Also so would it be:
first interval: 20 requests
second interval: 0 requests
third interval: 20 requests.

If the instance has 
timeValue = 2 seconds
and 
requestsAllowed = 20 requests

then when the third interval gets pushed it wouldn't work so it would return false.

Reply:
To clarify, what if we set timeValue = 1 second and requestsAllowed = 10 requests?

Reply:
oh i see.

Reply:
If its set to that. Then, it would return false after 10 requests.